src/librustc_lexer/src/lib.rs

```diff
@@ -6,78 +20,141 @@ pub mod unescape;
 #[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
 pub enum TokenKind {
+    // Multi-char tokens:
+
+    /// "// comment"
     LineComment,
+    /// "/* block comment */"
+    /// Block comments can be recursive, so the sequence like "/* /* */"
+    /// will not be considered terminated and will result in a parsing error.
     BlockComment { terminated: bool },
+    /// Any whitespace characters sequence.
     Whitespace,
+    /// "ident"
     Ident,
+    /// "r#ident"
     RawIdent,
+    /// "12_u8", "1.0e-40", "b"123"". See `LiteralKind` for more details.
     Literal { kind: LiteralKind, suffix_start: usize },
+    /// "'a"
     Lifetime { starts_with_number: bool },
+
+    // One-char tokens:
+
+    /// ";"
     Semi,
+    /// ","
     Comma,
+    /// "."
     Dot,
+    /// "("
     OpenParen,
+    /// ")"
     CloseParen,
+    /// "{"
     OpenBrace,
+    /// "}"
     CloseBrace,
+    /// "["
     OpenBracket,
+    /// "]"
     CloseBracket,
+    /// "@"
     At,
+    /// "#"
     Pound,
+    /// "~"
     Tilde,
+    /// "?"
     Question,
+    /// ":"
     Colon,
+    /// "$"
     Dollar,
+    /// "="
     Eq,
+    /// "!"
     Not,
+    /// "<"
     Lt,
+    /// ">"
     Gt,
+    /// "-"
     Minus,
+    /// "&"
     And,
+    /// "|"
     Or,
+    /// "+"
     Plus,
+    /// "*"
     Star,
+    /// "/"
     Slash,
+    /// "^"
     Caret,
+    /// "%"
     Percent,
+
+    /// Unknown token, not expected by the lexer, e.g. "№"
     Unknown,
 }
 use self::TokenKind::*;
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
 pub enum LiteralKind {
+    /// "12_u8", "0o100", "0b120i99"
     Int { base: Base, empty_int: bool },
+    /// "12.34f32", "0b100.100"
     Float { base: Base, empty_exponent: bool },
+    /// "'a'", "'\\'", "'''", "';"
     Char { terminated: bool },
+    /// "b'a'", "b'\\'", "b'''", "b';"
     Byte { terminated: bool },
+    /// ""abc"", ""abc"
     Str { terminated: bool },
+    /// "b"abc"", "b"abc"
     ByteStr { terminated: bool },
+    /// "r"abc"", "r#"abc"#", "r####"ab"###"c"####", "r#"a"
     RawStr { n_hashes: usize, started: bool, terminated: bool },
+    /// "br"abc"", "br#"abc"#", "br####"ab"###"c"####", "br#"a"
     RawByteStr { n_hashes: usize, started: bool, terminated: bool },
 }
 use self::LiteralKind::*;
 
+/// Base of numeric literal encoding according to its prefix.
 #[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
 pub enum Base {
+    /// Literal starts with "0b".
     Binary,
+    /// Literal starts with "0o".
     Octal,
+    /// Literal starts with "0x".
     Hexadecimal,
+    /// Literal doesn't contain a prefix.
     Decimal,
 }
 
-impl Token {
-    fn new(kind: TokenKind, len: usize) -> Token {
-        Token { kind, len }
-    }
-}
-
+/// Skips the line if it starts with a shebang ("#!").
+/// Line won't be skipped if it represents a valid Rust syntax
+/// (e.g. "#![deny(missing_docs)]").
 pub fn strip_shebang(input: &str) -> Option<usize> {
```

Comment:
Here and probably elsewhere, it's better to describe not what the code does, but *why* it does this. For example, here it makes sense to mention rustc allows hash-bang at the start of a file, to allow `#!/usr/bin/rustrun`. This gives context about how the function should be used.

Reply:
Well, at least I tried to do so ^_^". Here I just assumed that the purpose of shebang is pretty clear. However I agree, it makes sense to add such a note.